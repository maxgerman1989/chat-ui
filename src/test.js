import React, { Component } from "react";
import Message from "./components/message/message";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { MuiThemeProvider } from "material-ui/styles";
const puppeteer = require("puppeteer");

global.navigator = { userAgent: "all" };

//Snapshot of the message component
describe("Message component", () => {
  test("Message Component Snapshot", () => {
    let date = new Date(2018, 11, 2, 10).toLocaleString();
    const tree = renderer
      .create(
        <MuiThemeProvider>
          <Message
            avatar="https://spotim-demo-chat-server.herokuapp.com/avatars/001-snorlax.png"
            username="max"
            date={date}
            message="message"
            highlightMessage={true}
          />
        </MuiThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

//Testing the message input component to send a message to the server.
describe("Message Input component", () => {
  test("Can send message form", async () => {
    let browser = await puppeteer.launch({
      headless: false,
      devtools: true
    });
    let chatMessage = {
      username: "max",
      message: "hey!"
    };
    let page = await browser.newPage();

    page.emulate({
      viewport: {
        width: 500,
        height: 900
      },
      userAgent: ""
    });

    await page.goto("http://localhost:3000/");
    await page.waitForSelector(".messageForm");
    await page.click("input[name=username]");
    await page.type("input[name=username]", chatMessage.username);
    await page.click("input[name=message]");
    await page.type("input[name=message]", chatMessage.message);
    await page.click("button[type=submit]");

    browser.close();
  }, 9000000);
});
