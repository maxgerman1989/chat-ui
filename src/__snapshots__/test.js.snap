// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component Message Component Snapshot 1`] = `
<div
  className="ui card message highlight"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      style={
        Object {
          "MozBoxSizing": "border-box",
          "boxSizing": "border-box",
          "fontWeight": 500,
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    >
      <img
        size={40}
        src="https://spotim-demo-chat-server.herokuapp.com/avatars/001-snorlax.png"
        style={
          Object {
            "MozUserSelect": "none",
            "WebkitAlignItems": "center",
            "WebkitBoxAlign": "center",
            "WebkitBoxPack": "center",
            "WebkitJustifyContent": "center",
            "WebkitUserSelect": "none",
            "alignItems": "center",
            "backgroundColor": "rgb(188, 188, 188)",
            "borderRadius": "50%",
            "color": "#ffffff",
            "display": "inline-flex",
            "fontSize": 20,
            "height": 40,
            "justifyContent": "center",
            "marginRight": 16,
            "msFlexAlign": "center",
            "msFlexPack": "center",
            "msUserSelect": "none",
            "muiPrepared": true,
            "userSelect": "none",
            "width": 40,
          }
        }
      />
      <div
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "paddingRight": "90px",
            "verticalAlign": "top",
            "whiteSpace": "normal",
          }
        }
      >
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 15,
              "muiPrepared": true,
            }
          }
        >
          max
        </span>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "display": "block",
              "fontSize": 14,
              "muiPrepared": true,
            }
          }
        >
          2018-12-2 10:00:00
        </span>
      </div>
    </div>
    <div
      className="messageContent"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 14,
          "muiPrepared": true,
          "padding": 16,
        }
      }
    >
      message
    </div>
  </div>
</div>
`;
